quickwit-indexing/src/source/kinesis/kinesis_source.rs

```diff
@@ -340,16 +341,23 @@ impl Source for KinesisSource {
 }
 
 pub(super) fn get_region(region_or_endpoint: Option<RegionOrEndpoint>) -> anyhow::Result<Region> {
-    match region_or_endpoint {
-        Some(RegionOrEndpoint::Endpoint(endpoint)) => Ok(Region::Custom {
+    if let Some(RegionOrEndpoint::Endpoint(endpoint)) = region_or_endpoint {
+        return Ok(Region::Custom {
             name: "Custom".to_string(),
             endpoint,
-        }),
-        Some(RegionOrEndpoint::Region(region)) => region
-            .parse()
-            .with_context(|| format!("Failed to parse region: `{}`", region)),
-        None => Ok(Region::default()),
+        });
     }
+
+    // TODO: Change this to let_chain when 1.64 lands. (this is ugly)
+    if let Some(RegionOrEndpoint::Region(region)) = region_or_endpoint {
+        match region.parse() {
+            Ok(region_val) => return Ok(region_val),
+            Err(_) => {}
+        }
+    }
+
+    sniff_s3_region_and_cache() //< We fallback to S3 region if parsing from `region_or_endpoint`
```

Comment:
If parsing from the given region fails, it tries to get the region from `sniff_s3_region`. Should I keep it this way and add some logs to notify the user or should I revert the change to fail completely if it fails to resolve?